plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -46,32 +62,29 @@ class LintConfigurator implements Configurator {
         }
     }
 
-    private void configureToolTask() {
-        // evaluate violations after lint
-        def collectViolations = createCollectViolationsTask(violations)
+    private void configureCollectViolationsTask(variant = null) {
+        def taskSuffix = variant ? variant.name : ''
+        def collectViolations = createCollectViolationsTask(taskSuffix, violations).with {
+            it.dependsOn project.tasks.findByName("lint${taskSuffix.capitalize()}")
+        }
         evaluateViolations.dependsOn collectViolations
-        collectViolations.dependsOn project.tasks['lint']
     }
 
-    private CollectLintViolationsTask createCollectViolationsTask(Violations violations) {
-        project.tasks.create('collectLintViolations', CollectLintViolationsTask) { collectViolations ->
-            collectViolations.xmlReportFile = xmlOutputFile
-            collectViolations.htmlReportFile = new File(defaultOutputFolder, 'lint-results.html')
-            collectViolations.violations = violations
+    private CollectLintViolationsTask createCollectViolationsTask(String taskSuffix, Violations violations) {
+        project.tasks.create("collectLint${taskSuffix.capitalize()}Violations", CollectLintViolationsTask) { task ->
+            def reportSuffix = taskSuffix ? "-$taskSuffix" : ''
+            task.xmlReportFile = xmlOutputFileFor(reportSuffix)
+            task.htmlReportFile = new File(defaultOutputFolder, "lint-results${reportSuffix}.html")
+            task.violations = violations
         }
     }
 
-    private File getXmlOutputFile() {
-        project.android.lintOptions.xmlOutput ?: new File(defaultOutputFolder, 'lint-results.xml')
+    private File xmlOutputFileFor(reportSuffix) {
```

Comment:
I would prefer to push the report name directly and take the decision up two levels from here, so to keep it close to where we call `configureCollectViolationsTask()`

Reply:
I'm not sure what you mean by 2 level. Do mean the `with` clause above just after we create the task. The `xmlReportFile` property of the task is better to be configured while it is created.

Reply:
what I mean is that instead of adding this suffix logic in two places you can move the decision of deciding what task name and what report name to use up to L36 and L38.